source/features/stop-redirecting-commit-pull-url-to-commit-url.tsx

```diff
@@ -0,0 +1,31 @@
+import delegate from 'delegate-it';
+import * as pageDetect from 'github-url-detection';
+
+import features from '../libs/features';
+
+function init(): void {
+	delegate(document, 'form#new_issue', 'submit', updateTextArea);
+	delegate(document, 'form.js-new-comment-form', 'submit', updateTextArea);
+}
+
+function updateTextArea(event: delegate.Event): void {
+	const field = event.delegateTarget.querySelector('textarea');
+
+	field!.value = field!.value.replace(/\bhttps?:\/\/github.com\/.*\/pull\/.*\b/gi, (match): string => {
```

Comment:
1. This regex should only affect PR commit links, which look like USER/REPO/pull/NUMBER/commits/HASH
2. Reformatting isn’t necessary; to stop GitHub from changing the link you just need to add a space: `[url ](url)`
3. Replacement can be done without function, I think it’s just`.replace(regex, '[$1 ]($1))`
4. The regex needs to be created on the top level, after the imports, because it needs to use the current domain and not just location.origin (I can help you with this part)
5. At last, but most importantly, replacement can only be done once. This I can also help you with 

However we need this replacement function to be tested. Can you add tests for it? You can create this in libs/util.ts:

```js
function preventPrCommitLinkBreak(comment: string) {
  return comment.replace(…, …);
}
``` 

And then add a test in tests/utils.ts